### PR TITLE
[ready] `triangular_matrix_vector_product`: kokkos impl

### PR DESCRIPTION
**Pull request by fnrizzi**
#162   is a prerequisite for this, I can rebase once that is merged

### CODE REVIEW

File: tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/blas2_triangular_mat_vec_product.hpp [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+					Layout_x,
+					std::experimental::default_accessor<ElementType_x>
+				      > x,
+				      std::experimental::mdspan<ElementType_y,
+					std::experimental::extents<ext_y>,
+					Layout_y,
+					std::experimental::default_accessor<ElementType_y>
+				      > y)
+{
+  // constraints
```
Comment (mhoemmen) [inline tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/blas2_triangular_mat_vec_product.hpp:47]: Please see comments on other PRs about Constraints vs. Mandates; thanks!

File: tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/blas2_triangular_mat_vec_product.hpp [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+			 });
+    //fence message when using latest kokkos: "KokkosStdBlas::overwriting_triangular_mat_vec_upper: fence after operation"); 
+    ex.fence();
+  }
+
+  else{
+
+  // this print is detected in the tests
+#if defined KOKKOS_STDBLAS_ENABLE_TESTS
+  std::cout << "overwriting_triangular_matrix_vector_product_lower: kokkos impl\n";
```
Comment (mhoemmen) [inline tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/blas2_triangular_mat_vec_product.hpp:104]: Please see comments on my other PRs about logging encapsulation; thanks!

File: tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/mdspan_to_view_mapper_kk.hpp [hunk 3]
```diff
@@ -22,19 +30,72 @@ struct LayoutMapper<std::experimental::layout_right> {
 };
 
 template<class MDSpan>
-std::enable_if_t<MDSpan::rank()==1, Kokkos::View<typename MDSpan::element_type*>>
+std::enable_if_t<
+  MDSpan::rank()==1
+  && !is_complex_v<typename MDSpan::element_type>,
```
Comment (mhoemmen) [inline tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/mdspan_to_view_mapper_kk.hpp:35]: Please see my comments on PR #162.  That's the only thing I think _needs_ to be fixed in this PR.  Thanks!

**Review by mhoemmen — CHANGES_REQUESTED** [review]

**Review by mhoemmen — APPROVED** [review]

File: tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/blas2_triangular_mat_vec_product.hpp [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+    // ex.fence("KokkosStdBlas::updating_triangular_mat_vec_upper: fence after operation");
+    ex.fence();
+
+  }
+
+  else{
+
+    Impl::signal_kokkos_impl_called("updating_triangular_matrix_vector_product_lower");
+
+    Kokkos::parallel_for(Kokkos::RangePolicy(ex, 0, A_view.extent(0)),
```
[the remark this answers is not in the document]
Comment (fnrizzi) [reply tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/blas2_triangular_mat_vec_product.hpp:246]: not that I see! there is only `trmm` https://github.com/kokkos/kokkos-kernels/wiki/BLAS

**Pull request merged**